plugins/FiberDRCaloSDAction.cpp

```diff
@@ -0,0 +1,266 @@
+        }
+
+        return i;
+      }
+
+      // Linear interpolation function for calculating the efficiency of yellow filter, used in rejectedByYellowFilter
+      double getFilterEff(G4double G4energy) {
+        double energy = (G4energy * (dd4hep::MeV / CLHEP::MeV)); // Convert G4 MeV to dd4hep MeV
+        if (energy <= fGraph_X[0]) // If the photon energy <= 1.37760 eV, than return maximum filter efficiency
+          return fGraph_Y[0];
+        
+        for(int idx = 0; idx < fArraySize; ++idx) {
+          if (energy <= fGraph_X[idx]) {
+            double x1 = fGraph_X[idx - 1];
+            double x2 = fGraph_X[idx];
+            double y1 = fGraph_Y[idx - 1];
+            double y2 = fGraph_Y[idx];
+
+            return (y1 + ((y2 - y1) / (x2 - x1))*(energy - x1)); // return linear interpolated filter efficiency
+          }
+        }
+
+        // This should not happen
+        std::cout << "Error in Yellow filter efficiency with photon energy : " << energy << " MeV" << std::endl;
+        std::cout << "Cannot find corresponding filter efficiency" << std::endl;
+        return 0.;
+      }
+
+      // If true, then the photon is rejected by yellow filter
+      bool rejectedByYellowFilter(G4double G4energy, double rndVal)
+      {
+        double energy = (G4energy * (dd4hep::MeV / CLHEP::MeV)); // Convert G4 MeV to dd4hep MeV
+        if ( energy >= fGraph_X[fArraySize-1] ) // Photon w/ E > 2.69531 eV rejected
+          return true;
+
+        double FilterEff = getFilterEff(G4energy); // Get efficiency of filter using photon's energy
+
+        // filter efficiency == probability of photon accepted by filter
+        // == Probability of random value (from uniform distribution with range 0 ~ 1) smaller than filter efficiency
+        // So if the rndVal is larger than the FilterEff, than the photon is rejected
+        return (rndVal > FilterEff);
+      }
+
+      DRCData()
+          : fHitCollection(0), fHCID(-1), fWavBin(120), fTimeBin(650),
+            fWavlenStart(900.), fWavlenEnd(300.), fTimeStart(5.), fTimeEnd(70.)
+      {
+        fWavlenStep = (fWavlenStart - fWavlenEnd) / (float)fWavBin;
+        fTimeStep = (fTimeEnd - fTimeStart) / (float)fTimeBin;
+      }
+    };
+
+    /// Define collections created by this sensitive action object
+    template <>
+    void Geant4SensitiveAction<DRCData>::defineCollections()
+    {
+      std::string readout_name = m_sensitive.readout().name();
+      m_collectionID = defineCollection<Geant4DRCalorimeter::Hit>(m_sensitive.readout().name());
+      std::cout << "defineCollection Geant4DRCalorimeter readout_name   : " << readout_name << std::endl;
+      std::cout << "defineCollection Geant4DRCalorimeter m_collectionID : " << m_collectionID << std::endl;
+    }
+
+    /// Method for generating hit(s) using the information of G4Step object.
+    template <>
+    G4bool Geant4SensitiveAction<DRCData>::process(G4Step GEANT4_CONST_STEP *step, G4TouchableHistory *history)
```

Comment:
```suggestion
    G4bool Geant4SensitiveAction<DRCData>::process(G4Step GEANT4_CONST_STEP *step, G4TouchableHistory*)
```

Currently I still see a warning